src/Microsoft.OData.Client/Serialization/Serializer.cs

```diff
@@ -447,8 +446,9 @@ private void WriteODataId(LinkDescriptor linkDescriptor, ODataWriterWrapper oDat
         /// </summary>
         /// <param name="entityDescriptor">The entity descriptor.</param>
         /// <param name="bulkUpdateGraph">An instance of the <see cref="BulkUpdateGraph"/></param>
+        /// <param name="linkDescriptors">Descriptor links for some descriptors.</param>
         /// <param name="oDataWriter">The writer.</param>
-        internal void WriteDeltaEntry(EntityDescriptor entityDescriptor, BulkUpdateGraph bulkUpdateGraph, ODataWriterWrapper oDataWriter)
+        internal void WriteDeltaEntry(EntityDescriptor entityDescriptor, BulkUpdateGraph bulkUpdateGraph, Dictionary<Descriptor, LinkDescriptor> linkDescriptors, ODataWriterWrapper oDataWriter)

```

Comment:
Curious why we are passing `linkDescriptors` yet the BulkUpdateRequest was working without it

Reply:
@ElizabethOkerio Could you confirm that the following scenarios involving updated entities at a nested level greater than 1 are supported:

```csharp
// Scenario 1 - collection-valued navigation property
var customer1 = dataServiceContext.CreateQuery<Customer>("Customers(1)").AddQueryOption("$expand", "Orders($expand=Items)");

var order = customer1.Orders[0];
var orderItem = order.Items[0];
orderItem.Quantity = 3;

dataServiceContext.UpdateObject(orderItem);
dataServiceContext.BulkUpdate(customer1);


// Scenario 2 - single-valued navigation property
var order1 = dataServiceContext.CreateQuery<Order>("Orders(1)").AddQueryOption("$expand", "Customer($expand=RelationshipManager)");

var customer = order1.Customer;
var relationshipManager = customer.RelationshipManager;
relationshipManager.Name = "Sue";

dataServiceContext.UpdateObject(relationshipManager);
dataServiceContext.BulkUpdate(order1);
```

Reply:
[       public void DeepUpdate_WithTwoLevelsOfNesting_ShouldBeSerializedSuccessfully()](https://github.com/OData/odata.net/blob/3448900f1bfface016aeeb27fd311c0747a44b4f/test/FunctionalTests/Tests/DataServices/UnitTests/Client.TDD.Tests/Tests/BulkUpdateE2ETests.cs#L373)